document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -40,69 +36,42 @@
 
     private static final String SMALL_FULL = "smallfull";
 
+    private static final String ABRIDGED_REGEX = "/transactions\\/[0-9-]+/accounts\\/.*";
+
+    private static final String SMALL_FULL_REGEX = "/transactions\\/[0-9-]+/company-accounts\\/.*";
+
     private static final String ERROR_CALLING_ACCOUNTS = "An error occurred when calling: %s handler to obtain: %s" +
-             " accounts data for transaction: %s and resource link: %s";
+             " accounts data for resourceUri: %s";
 
     @Override
     public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) throws DocumentInfoException {
 
-        String resourceId = documentInfoRequest.getResourceId();
         String resourceUri = documentInfoRequest.getResourceUri();
         String requestId = documentInfoRequest.getRequestId();
 
         final Map< String, Object > debugMap = new HashMap< >();
         debugMap.put("resource_uri", resourceUri);
-        debugMap.put("resource_id", resourceId);
 
         LOG.infoContext(requestId,"Started getting document info", debugMap);
 
-        Transaction transaction;
-        try {
-            transaction = transactionService.getTransaction(resourceId, requestId);
-        } catch (ServiceException e) {
-            LOG.errorContext(requestId,"An error occurred when calling the transaction service with resource id: "
-                    + resourceId, e, debugMap);
-            throw new DocumentInfoException("Failed to get transaction with resourceId: " + resourceId, e);
-        }
-
-        String resourceLink =  Optional.of(transaction)
-                .map(Transaction::getResources)
-                .map(resources -> resources.get(resourceUri))
-                .map(Resources::getLinks)
-                .map(links -> links.get(LinkType.RESOURCE.getLink()))
-                .orElseGet(() -> {
-                    LOG.infoContext(requestId,"Unable to find resource: " + resourceId
-                            + " in transaction: " + resourceUri, debugMap);
-                    return "";
-                });
-
-
-        // Will need to be refactored when additional accounts are added.
         String accountType = "";
         try {
-            if (isAbridgedAccounts(resourceLink)) {
+            if (resourceUri.matches(ABRIDGED_REGEX)) {
```

Comment:
A factory could be used to retrieve the required handler. This could become difficult to read with more account types added.

Reply:
100% agree, I can change it. But, I was following the path of the original accounts Implemented which just checked if it was one or the other, as this was meant to be small changes to allow the resourceUri to be used.

Reply:
A factory would no longer be needed as we are now using either abridged or company account there is only 2 paths, with future changes of moving abridged into company accounts there will be only 1 path.